### PR TITLE
Add icons and remove empty recommendation sections

### PR DESCRIPTION
**Pull request by Hyperkid123**
### Changes
- added icons (for items that we have the data)
- changed default recommendation icon color to gray
- hide recommendation titles for with no children

screenshot-prod foo redhat com_1337-2021 04 08-11_28_09

### CODE REVIEW

File: src/components/app-content-renderer/recommendation-tile.js [hunk 1]
```diff
@@ -21,30 +13,28 @@ import {
...
 
 const RecommendationGroup = (recommendation) => {
   const intl = useIntl();
+  const dispatch = useDispatch();
   const [{ count, response, show }] = useRequest(recommendation);
 
+  useEffect(() => {
+    if (show === false) {
+      dispatch(
+        removeRecommendationTile(recommendation.id, recommendation.recId)
```
Comment (rvsia) [inline src/components/app-content-renderer/recommendation-tile.js:28]: You can use `onRequest` argument in the hook to do this. See `FirstPanelTile`.

Comment (Hyperkid123) [reply src/components/app-content-renderer/recommendation-tile.js:28]: Yeah, we will also need the on error callback with the same code. I've realized it yesterday evening.

Comment (rvsia) [reply src/components/app-content-renderer/recommendation-tile.js:28]: Yeah, that's the third argument `onError` for! 😎

File: src/components/app-content-renderer/recommendation-tile.js [hunk 2]
```diff
@@ -21,28 +13,28 @@ import {
...
   const intl = useIntl();
-  const [{ count, response, show }] = useRequest(recommendation);
+  const dispatch = useDispatch();
+  const removeTile = ({ show }) =>
+    show === false &&
+    dispatch(removeRecommendationTile(recommendation.id, recommendation.recId));
+  const [{ count, response, show }] = useRequest(
+    recommendation,
+    removeTile,
+    () => removeTile({ show: false })
```
Comment (rvsia) [inline src/components/app-content-renderer/recommendation-tile.js:29]: should be true I think

File: src/components/app-content-renderer/recommendation-tile.js [hunk 3]
```diff
@@ -1,15 +1,7 @@
 import React from 'react';
-import PropTypes from 'prop-types';
+import PropTypes, { string } from 'prop-types';
```
Comment (rvsia) [inline src/components/app-content-renderer/recommendation-tile.js:2]: probably unintentional import

File: src/components/app-content-renderer/recommendation-tile.js (hunk 2 above)
Comment (Hyperkid123) [reply src/components/app-content-renderer/recommendation-tile.js:29]: No, it should be false, we want to remove tiles on API errors.

Comment (rvsia) [reply src/components/app-content-renderer/recommendation-tile.js:29]: oh, sorry! 🤦

**Pull request merged**